tests/karate/src/test/java/be/utils/JsonConverter.java

```diff
@@ -0,0 +1,143 @@
+  private String senderSk = "0leCDBokllJXKXT72psnqF5UYFVRxnc1BNDShY05KHQn18HMlXvQmTlz6LfbvtSrgKZ4vi3ndYN9SCForQel4w==";
+
+  private String signature = "ONylxgHA9cbsB_lwdfbn3iyzRd4aTpJhBMnvEKhmJF_niE_pUHdmjxDXjEwFyvo5WiH1NZXWyXG27SYEpkasCA==";
+  private String messageIdForced = "";
+  /*
+    Produces the base64 variant of the json file passed as argument
+   */
+  public String convertJson(Json json){
+    String stringJson = json.toString();
+    byte[] jsonBytes = stringJson.getBytes(StandardCharsets.UTF_8);
+    Base64.Encoder encoder = Base64.getEncoder();
+    String base64 = encoder.encodeToString(jsonBytes);
+    return base64;
+  }
+
+  /*
+    Produces a valid Json representation of a message given the message data, the id of the message
+    and the channel where the message is supposed to be sent
+   */
+  public Json publishМessageFromData(String stringData, int id, String channel){
+    Json messageData = Json.of(stringData);
+    String messageDataBase64 = convertJson(messageData);
+    System.out.println("data : "+messageDataBase64+" type: "+ "publish"+ " id : "+ id+ "channel : "+ channel);
+    Map<String,Object> messageJson = new LinkedHashMap<>();
+    messageJson.put("method","publish");
+    messageJson.put("id",id);
+    Map<String,Object> paramsPart = constructParamsField(channel,messageDataBase64);
+    messageJson.put("params",paramsPart);
+    messageJson.put("jsonrpc","2.0");
+    return Json.of(messageJson);
+  }
+
+  public Map<String,Object> constructMessageField(String messageDataBase64, boolean forcedMessageId){
+    Map<String,Object> messagePart  = new LinkedHashMap<>();
+    messagePart.put("data",messageDataBase64);
+    messagePart.put("sender",senderPk);
+    String signature = constructSignature(messageDataBase64);
+    messagePart.put("signature",signature);
+    String messageId = hash(messageDataBase64.getBytes(StandardCharsets.UTF_8),signature.getBytes(StandardCharsets.UTF_8));
+    if(forcedMessageId && !messageIdForced.isEmpty()){
+      messageId = this.messageIdForced;
+      messageIdForced = "";
+    }
+    messagePart.put("message_id",messageId);
+    System.out.println("message id is : "+messageId);
+    String[] witness = new String[0];
+    messagePart.put("witness_signatures",witness);
+    return messagePart;
+  }
+
+  public Map<String,Object> constructParamsField(String channel, String messageDataBase64){
+    Map <String,Object> paramsPart = new LinkedHashMap<>();
+    paramsPart.put("channel",channel);
+
+    Map<String,Object> messagePart = constructMessageField(messageDataBase64,false);
+    paramsPart.put("message",messagePart);
+
+    return paramsPart;
+  }
+
+  /*
+      Constructs a valid signature on given data
+   */
+  public String constructSignature(String messageDataBase64){
+    try {
+      // Hex representation of the private key
+      String privateKeyHex = "1498b5467a63dffa2dc9d9e069caf075d16fc33fdd4c3b01bfadae6433767d93";
```

Comment:
This is bad practice.

This should be set as constant.

But what I don't understand is how `senderPk` can be mutable while the private key is fixed.
Isn't that an issue ?